provider.go

```diff
@@ -74,15 +166,28 @@ func getAPI(hostAddr string, maxRetries int) (client.API, error) {
 func Provider() terraform.ResourceProvider {
 	return &schema.Provider{
 		Schema: map[string]*schema.Schema{
-			"tunnel_address": &schema.Schema{
+			"driver": &schema.Schema{
 				Type:     schema.TypeString,
-				Required: true,
+				Optional: true,
+				Default: "tunnel",
+				Description: "Driver to use to connect to Fleet. Can be tunnel or api.",
```

Comment:
Modify all you want. I doubt many people are using this provider, so it's probably more important that we get it right than it is to maintain backwards compatibility.

I think we should more or less be able to use the code from https://github.com/coreos/fleet/blob/master/fleetctl/fleetctl.go with some modifications to get the configuration from Terraform instead of `globalFlags`. It's a shame it's not exported in a more easily reused way, so I guess some copy & paste could be fine. :disappointed: